x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -308,6 +284,48 @@ export function App({
             />
           )}
         </div>
+        {lastKnownDoc && state.isSaveModalVisible && (
+          <SavedObjectSaveModal
+            onSave={props => {
+              const doc = {
+                ...lastKnownDoc,
+                id: props.newCopyOnSave ? undefined : lastKnownDoc.id,
+                title: props.newTitle,
+              };
+
+              docStorage
+                .save(doc)
+                .then(({ id }) => {
+                  // Prevents unnecessary network request and disables save button
+                  const newDoc = { ...doc, id };
+                  setState(s => ({
+                    ...s,
+                    isSaveModalVisible: false,
+                    persistedDoc: newDoc,
+                    lastKnownDoc: newDoc,
+                  }));
+
+                  if (docId !== id) {
+                    redirectTo(id);
+                  }
+                })
+                .catch(() => {
+                  core.notifications.toasts.addDanger(
+                    i18n.translate('xpack.lens.app.docSavingError', {
+                      defaultMessage: 'Error saving document',
+                    })
+                  );
+                  setState(s => ({ ...s, isSaveModalVisible: false }));
+                });
+            }}
+            onClose={() => setState(s => ({ ...s, isSaveModalVisible: false }))}
+            title={lastKnownDoc.title || ''}
```

Comment:
This is a weird experience because the default title is "New visualization", but I'd rather have the default title be blank. With a default title I need to delete the existing text before typing, which is annoying.